pre-release CI failures for `signal.windows`

These 4 test failures are visible in several jobs, and are caused by https://github.com/numpy/numpy/pull/23399:

```
___________________________ TestBartHann.test_basic ____________________________
[gw0] linux -- Python 3.9.16 /opt/hostedtoolcache/Python/3.9.16/x64/bin/python
scipy/signal/tests/test_windows.py:44: in test_basic
    assert_allclose(windows.barthann(6, sym=True),
        self       = <scipy.signal.tests.test_windows.TestBartHann object at 0x7fbbb0f2c550>
/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=1e-07, atol=0
E   
E   Mismatched elements: 2 / 6 (33.3%)
E   Max absolute difference: 1.11022302e-16
E   Max relative difference: 1.54811063e-16
E    x: array([1.110223e-16, 3.585735e-01, 8.794265e-01, 8.794265e-01,
E          3.585735e-01, 1.110223e-16])
E    y: array([0.      , 0.358574, 0.879426, 0.879426, 0.358574, 0.      ])
        args       = (<function assert_allclose.<locals>.compare at 0x7fbb83b30c10>, array([1.11022302e-16, 3.58573542e-01, 8.79426458e-01,...58573542e-01, 1.11022302e-16]), array([0.        , 0.35857354, 0.87942646, 0.87942646, 0.35857354,
       0.        ]))
        func       = <function assert_array_compare at 0x7fbbc6117040>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=1e-07, atol=0', 'verbose': True}
        self       = <contextlib._GeneratorContextManager object at 0x7fbbc6111d00>
_____________________________ TestHann.test_basic ______________________________
[gw0] linux -- Python 3.9.16 /opt/hostedtoolcache/Python/3.9.16/x64/bin/python
scipy/signal/tests/test_windows.py:402: in test_basic
    assert_allclose(windows.hann(6, sym=False),
        self       = <scipy.signal.tests.test_windows.TestHann object at 0x7fbbb0e4bac0>
/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=1e-07, atol=0
E   
E   Mismatched elements: 1 / 6 (16.7%)
E   Max absolute difference: 3.33066907e-16
E   Max relative difference: 1.33226763e-15
E    x: array([1.110223e-16, 2.500000e-01, 7.500000e-01, 1.000000e+00,
E          7.500000e-01, 2.500000e-01])
E    y: array([0.  , 0.25, 0.75, 1.  , 0.75, 0.25])
        args       = (<function assert_allclose.<locals>.compare at 0x7fbb81239af0>, array([1.11022302e-16, 2.50000000e-01, 7.50000000e-01, 1.00000000e+00,
       7.50000000e-01, 2.50000000e-01]), array([0.  , 0.25, 0.75, 1.  , 0.75, 0.25]))
        func       = <function assert_array_compare at 0x7fbbc6117040>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=1e-07, atol=0', 'verbose': True}
        self       = <contextlib._GeneratorContextManager object at 0x7fbbc6111d00>
_____________________________ TestTukey.test_basic _____________________________
[gw0] linux -- Python 3.9.16 /opt/hostedtoolcache/Python/3.9.16/x64/bin/python
scipy/signal/tests/test_windows.py:565: in test_basic
    assert_allclose(win, v, rtol=1e-14)
        k          = (4, 0.5, True)
        self       = <scipy.signal.tests.test_windows.TestTukey object at 0x7fbbb0e4b5b0>
        v          = array([0., 1., 1., 0.])
        win        = array([1.11022302e-16, 1.00000000e+00, 1.00000000e+00, 1.11022302e-16])
/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=1e-14, atol=0
E   
E   Mismatched elements: 2 / 4 (50%)
E   Max absolute difference: 1.11022302e-16
E   Max relative difference: 0.
E    x: array([1.110223e-16, 1.000000e+00, 1.000000e+00, 1.110223e-16])
E    y: array([0., 1., 1., 0.])
        args       = (<function assert_allclose.<locals>.compare at 0x7fbb81239d30>, array([1.11022302e-16, 1.00000000e+00, 1.00000000e+00, 1.11022302e-16]), array([0., 1., 1., 0.]))
        func       = <function assert_array_compare at 0x7fbbc6117040>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=1e-14, atol=0', 'verbose': True}
        self       = <contextlib._GeneratorContextManager object at 0x7fbbc6111d00>
_________________________ test_cosine_logpdf_endpoints _________________________
[gw0] linux -- Python 3.9.16 /opt/hostedtoolcache/Python/3.9.16/x64/bin/python
scipy/stats/tests/test_distributions.py:8763: in test_cosine_logpdf_endpoints
    assert_equal(logp, [-np.inf, -np.inf])
        logp       = array([-37.88153046, -37.88153046])
/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Arrays are not equal
E   
E   x and y -inf location mismatch:
E    x: array([-37.88153, -37.88153])
E    y: array([-inf, -inf])
        args       = (<built-in function eq>, array([-37.88153046, -37.88153046]), [-inf, -inf])
        func       = <function assert_array_compare at 0x7fbbc6117040>
        kwds       = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}
        self       = <contextlib._GeneratorContextManager object at 0x7fbbc6111d00>
```
The `stats.cosine` one was fixed by gh-18382 already, so there are now 3 failures left in `signal.windows`.

--- a/scipy/signal/tests/test_windows.py
+++ b/scipy/signal/tests/test_windows.py
@@ -43,11 +43,14 @@ class TestBartHann:
     def test_basic(self):
         assert_allclose(windows.barthann(6, sym=True),
                         [0, 0.35857354213752, 0.8794264578624801,
-                         0.8794264578624801, 0.3585735421375199, 0])
+                         0.8794264578624801, 0.3585735421375199, 0],
+                        rtol=1e-15, atol=1e-15)
         assert_allclose(windows.barthann(7),
-                        [0, 0.27, 0.73, 1.0, 0.73, 0.27, 0])
+                        [0, 0.27, 0.73, 1.0, 0.73, 0.27, 0],
+                        rtol=1e-15, atol=1e-15)
         assert_allclose(windows.barthann(6, False),
-                        [0, 0.27, 0.73, 1.0, 0.73, 0.27])
+                        [0, 0.27, 0.73, 1.0, 0.73, 0.27],
+                        rtol=1e-15, atol=1e-15)
 
 
 class TestBartlett:
@@ -400,16 +403,20 @@ class TestHann:
 
     def test_basic(self):
         assert_allclose(windows.hann(6, sym=False),
-                        [0, 0.25, 0.75, 1.0, 0.75, 0.25])
+                        [0, 0.25, 0.75, 1.0, 0.75, 0.25],
+                        rtol=1e-15, atol=1e-15)
         assert_allclose(windows.hann(7, sym=False),
                         [0, 0.1882550990706332, 0.6112604669781572,
                          0.9504844339512095, 0.9504844339512095,
-                         0.6112604669781572, 0.1882550990706332])
+                         0.6112604669781572, 0.1882550990706332],
+                        rtol=1e-15, atol=1e-15)
         assert_allclose(windows.hann(6, True),
                         [0, 0.3454915028125263, 0.9045084971874737,
-                         0.9045084971874737, 0.3454915028125263, 0])
+                         0.9045084971874737, 0.3454915028125263, 0],
+                        rtol=1e-15, atol=1e-15)
         assert_allclose(windows.hann(7),
-                        [0, 0.25, 0.75, 1.0, 0.75, 0.25, 0])
+                        [0, 0.25, 0.75, 1.0, 0.75, 0.25, 0],
+                        rtol=1e-15, atol=1e-15)
 
 
 class TestKaiser:
@@ -562,7 +569,7 @@ class TestTukey:
                 assert_raises(ValueError, windows.tukey, *k)
             else:
                 win = windows.tukey(*k)
-                assert_allclose(win, v, rtol=1e-14)
+                assert_allclose(win, v, rtol=1e-15, atol=1e-15)
 
     def test_extremes(self):
         # Test extremes of alpha correspond to boxcar and hann

